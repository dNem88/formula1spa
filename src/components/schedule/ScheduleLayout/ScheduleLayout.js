import React, {Fragment, useEffect, useState} from 'react'
import ScheduleList from '../ScheduleList/ScheduleList'
import styles from './scheduleLayout.module.css'
import ErrorComp from '../../common/errorComp/ErrorComp'

import transformSchedule from '../../../utils/common/transformSchedule'
import schedule from '../../../utils/api/heroku_api/schedule'
import CommonHeader from '../../common/titleHeader/CommonHeader'
const headerProps = {
    header: 'F1 Schedule 2022',
    description: 'A record 23-race Formula 1 calendar for 2022, including the inaugural Miami Grand Prix*. Click on any Grand Prix for full F1 schedule details, dates, times & full weekend program. *Subject to circuit homologation.'
}
function ScheduleLayout() {

    const [state, setState] = useState({fullSchedule: null, error: null, errorMessage: null})


    useEffect(() => {
        async function FetchSchedule() {
            try {
                let response = await schedule.getFullSchedule()
                if (!response.ok) {
                    throw new Error('Failed to fetch fullSchedule!')
                }
                let json = await response.json()
                if (json.length < 20) {
                    throw new Error('Failed to get schedule! Some of the tracks are missing!')
                }
                const transformed = json.map(transformSchedule)
                setState({error: null, errorMessage: null, fullSchedule: transformed,})
            } catch(err) {
                setState({fullSchedule: null, error: true, errorMessage: err.message})
            }
        }
        FetchSchedule()
    }, [])
 
    return (
        <div className={styles['main-container']}>
             <CommonHeader {...headerProps} />
            {state.error ? <ErrorComp errorMessage={state.errorMessage}/> : null}
            {state.fullSchedule ? 
               <ScheduleList fullSchedule={state.fullSchedule}/> : null}
        </div>
    )
}

export default ScheduleLayout