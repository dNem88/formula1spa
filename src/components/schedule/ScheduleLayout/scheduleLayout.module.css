.main-container{
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-evenly;
    padding: 2vw;
}