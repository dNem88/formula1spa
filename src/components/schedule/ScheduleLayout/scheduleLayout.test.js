import React from 'react'
import renderer from 'react-test-renderer'
import ScheduleLayout from './ScheduleLayout'
import {render, screen} from '@testing-library/react'


describe('Test ScheduleLayout',() => {
    it('Match Snapshot', () => {
        const snapshot = renderer.create(
            <ScheduleLayout />
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.props.className).toBe('main-container')
    })
    it('Elements text content', () => {
        render(<ScheduleLayout />)
        expect(screen.getByText('F1 Schedule 2022')).toBeTruthy()
        expect(screen.getByText('A record 23-race Formula 1 calendar for 2022, ', {exact: false})).toBeTruthy()
    })
})