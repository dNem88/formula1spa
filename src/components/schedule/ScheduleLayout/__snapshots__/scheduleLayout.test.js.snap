// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ScheduleLayout Match Snapshot 1`] = `
<div
  className="main-container"
>
  <div
    className="wrapper"
  >
    <div
      className="header"
    >
      <h1>
        F1 Schedule 2022
      </h1>
    </div>
    <div
      className="description"
    >
      <p>
        A record 23-race Formula 1 calendar for 2022, including the inaugural Miami Grand Prix*. Click on any Grand Prix for full F1 schedule details, dates, times & full weekend program. *Subject to circuit homologation.
      </p>
    </div>
  </div>
  <div
    className="lds-ring"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
